### PR TITLE
DW-3509 - Adding self signed SSL cert

### PR DESCRIPTION
**Pull request by benj-fletch**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,18 +13,20 @@ To build the Docker image to a local Docker daemon, run the following from the r
...
-    dependsOn(":jibDockerBuild")
-}
+./gradlew clean build
+docker build --tag dwpdigital/emr-cluster-broker:latest .
 ```
 
 To start the image:
 ```
-docker run -p 8080:8080  -e AWS_REGION=<VALUE> -e AMI_SEARCH_PATTERN=<VALUE> -e AMI_OWNER_IDS=<VALUE> -e EMR_RELEASE_LABEL=<VALUE> -e S3_LOG_URI=<VALUE> -e SECURITY_CONFIGURATION=<VALUE> -e JOB_FLOW_ROLE_BLACKLIST=<VALUE>   AWS_REGION="testRegion" dwpdigital/emr-cluster-broker
+docker run -p 8080:8080 -e clusterBroker_awsRegion=<VALUE> -e clusterBroker_amiSearchPattern=<VALUE> -e clusterBroker_amiOwnerIds=<VALUE> -e clusterBroker_emrReleaseLabel=<VALUE> -e clusterBroker_s3LogUri=<VALUE> -e clusterBroker_securityConfiguration=<VALUE> -e clusterBroker_jobFlowRoleBlacklist=<VALUE> dwpdigital/emr-cluster-broker:latest
```
Comment (mattburgess) [inline README.md]: I like the env var prefixes (very 12 factor!), but don't like the mixed case. Env vars are generally all uppercase, so I've a preference for that if it's not going to cause issues for some reason?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -6,3 +6,9 @@ management.metrics.export.prometheus.enabled=true
 management.endpoints.web.base-path=/
 management.endpoints.web.path-mapping.prometheus=metrics
 management.endpoints.web.path-mapping.metrics=act-metrics
+
+# HTTPS config
+server.ssl.key-store-type=JKS
+server.ssl.key-store=/opt/emr-cluster-broker/emr-cluster-broker.keystore
+server.ssl.key-store-password=changeit
```
Comment (mattburgess) [inline src/main/resources/application.properties]: how does one override this password? I think we need to consider integrating https://github.com/dwp/acm-pca-cert-generator so we can pull down a static cert for non-local environments and put a randomised password on the keystores & truststores.

Comment (caseyr232) [reply src/main/resources/application.properties]: We had agreement of doing it this way from Jim, the main goal here is to just ensure the traffic between the ALB and container is encrypted, as long as we are generating a random certificate on startup we satisfy this requirement. The addition of ACM adds unneeded complexity (and bloat to the image).

File: README.md (hunk 1 above)
Comment (benj-fletch) [reply README.md]: These actually map through to Spring boot properties when the application is running. The mandate there is that they have to match the ones stored in `ConfigurationService` which uses the spring convention of camelCase and `.`'s. Whichever convention we pick would be breaking the other so no nice solution here!

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -6,3 +6,9 @@ management.metrics.export.prometheus.enabled=true
 management.endpoints.web.base-path=/
 management.endpoints.web.path-mapping.prometheus=metrics
 management.endpoints.web.path-mapping.metrics=act-metrics
+
+# HTTPS config
```
Comment (caseyr232) [inline src/main/resources/application.properties]: Conventionally we would usually switch the port to 8443 from the default of 8080 to denote HTTPs traffic, could you please add in the serverport change too?

Comment (benj-fletch) [reply src/main/resources/application.properties]: Sure, I will change this.

File: src/main/resources/application.properties (hunk 2 above)
Comment (benj-fletch) [reply src/main/resources/application.properties]: I wouldn't be opposed to doing this in future as a further hardening technique, though for now I think it is sufficient to generate the cert on container runtime as this has been agreed.
To be fully secure I think using the mentioned library is a good idea but the risk of this password is negligible.

File: README.md (hunk 1 above)
Comment (mattburgess) [reply README.md]: Boo :( Fine, let's leave them as is then.

File: src/main/resources/application.properties (hunk 2 above)
Comment (mattburgess) [reply src/main/resources/application.properties]: Hey, as long as Jim's happy, I'm happy.

**Pull request merged**